js/oceanex.js

```
@@ -0,0 +1,323 @@
+        }
+        return {
+            'info': trade,
+            'timestamp': trade['created_on'],
+            'datetime': trade['created_at'],
+            'symbol': trade['market'],
+            'id': trade['id'],
+            'order': undefined,
+            'type': 'limit',
+            'takerOrMaker': undefined,
+            'side': side,
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'volume'),
+            'cost': undefined,
+            'fee': undefined,
+        };
+    }
+
+    async fetchTime (params = {}) {
+        let response = await this.publicGetTimestamp ();
+        return response['data'];
+    }
+
+    async fetchTradingFees (params = {}) {
+        let response = await this.publicGetFeesTrading ();
+        return response['data'];
+    }
+
+    async fetchKey (params = {}) {
+        // private method input value {'user_jwt': encrypted_jwt_token}
+        // jwt_token should be RS256 encoded
+        let response = await this.privateGetKey (this.extend (params));
+        return response['data'];
+    }
+
+    async fetchBalance (params = {}) {
+        let response = await this.privateGetMembersMe (this.extend (params));
+        const balances = this.safeValue (response['data'], 'accounts');
+        let result = { 'info': balances };
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            let currency = balance['currency'];
+            let account = this.account ();
+            let free = this.safeFloat (balance, 'balance');
+            let used = this.safeFloat (balance, 'locked');
+            let total = free + used;
+            account['free'] = free;
+            account['used'] = used;
+            account['total'] = total;
+            result[currency] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
```

We should throw an exception from here if the exchange supports limit orders only:

- https://github.com/ccxt/ccxt/blob/master/js/upbit.js#L835
- https://github.com/ccxt/ccxt/pull/5091/files#diff-27adc3166fcc2d328d8e1ac942896cb0R177